src/main/java/net/pms/platform/windows/WindowsUtils.java

```diff
@@ -108,37 +105,21 @@ public File getKLiteFiltersDir() {
 	}
 
 	@Override
-	public String getShortPathNameW(String longPathName) {
-		if (longPathName == null) {
-			return null;
-		}
-		boolean unicodeChars;
-		try {
-			byte[] b1 = longPathName.getBytes(StandardCharsets.UTF_8);
-			byte[] b2 = longPathName.getBytes("cp1252");
-			unicodeChars = b1.length != b2.length;
-		} catch (UnsupportedEncodingException e) {
-			return longPathName;
-		}
-
-		if (unicodeChars) {
-			try {
-				WString pathname = new WString(longPathName);
-
-				char[] test = new char[2 + pathname.length() * 2];
-				int r = Kernel32.INSTANCE.GetShortPathNameW(pathname, test, test.length);
-				if (r > 0) {
-					String result = Native.toString(test);
-					LOGGER.trace("Using short path name of \"{}\": \"{}\"", pathname, result);
-					return result;
+	public String getSystemPathName(String path) {
+		if (path != null) {
+			File file = new File(path);
+			String resolved = file.getAbsolutePath();
+			if (resolved.length() > 247) {
```

Comment:
Why 247?

Reply:
On Windows the maximum short path is 260 minus 1 (`NUL`) -> 259 char.
But for directories it is 260 minus 12 (8.3 file) minus 1 (`NUL`) to allow for the creation of a file in the directory -> 247.

The long path prefix (`\\?\`) is working with short path as well.
For example, a file like `C:\ProgramData\UMS\UMS.conf` can have `\\?\C:\ProgramData\UMS\UMS.conf` as working path even if it is not more than 259 char (the max for short file path).

So it is no need to check if it is a directory or a file, just prefix as long path if it is more than 247 char.

Reply:
Interesting, thanks for explaining. I'll add that explanation as a code comment for future people